lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -1108,33 +1109,33 @@ std::string handshake::MemoryOp::getResultName(unsigned int idx) {
   return getMemoryResultName(ldCount(), stCount(), idx);
 }
 
-static LogicalResult verifyMemoryOp(handshake::MemoryOp op) {
-  auto memrefType = op.memRefType();
+LogicalResult handshake::MemoryOp::verify() {
+  auto memrefType = memRefType();
 
   if (memrefType.getNumDynamicDims() != 0)
-    return op.emitOpError()
+    return emitOpError()
            << "memref dimensions for handshake.memory must be static.";
   if (memrefType.getShape().size() != 1)
-    return op.emitOpError() << "memref must have only a single dimension.";
+    return emitOpError() << "memref must have only a single dimension.";
 
-  unsigned stCount = op.stCount();
-  unsigned ldCount = op.ldCount();
+  unsigned stCount = this->stCount();
+  unsigned ldCount = this->ldCount();
```

Comment:
Most places you don't bother with `this->`

Reply:
> Most places you don't bother with `this->`

Yes we can remove `op` in most case, 
here raise a compiler error.

```
../lib/Dialect/Handshake/HandshakeOps.cpp:1121:29: error: called object type 'unsigned int' is not a function or function pointer
  unsigned stCount = stCount();
                     ~~~~~~~^
../lib/Dialect/Handshake/HandshakeOps.cpp:1122:29: error: called object type 'unsigned int' is not a function or function pointer
  unsigned ldCount = ldCount();
```
likely because the variable and function have the same name `stCount` `ldCount` here.